File: javascript/resistor-color-duo/resistor-color-duo.js
const COLORS = [
  "black",
  "brown",
  "red",
  "orange",
  "yellow",
  "green",
  "blue",
  "violet",
  "grey",
  "white"
];

export const value = colors => {
  let res = 0;

  colors
    .slice()
    .reverse()
    .forEach((v, i) => {
      let id = COLORS.indexOf(v);

      res += id * 10 ** i;
    });

  return res;
};

File: ruby/resistor-color-duo/resistor_color_duo.rb
# frozen_string_literal: true

COLORS = {
  black: '0',
  brown: '1',
  red: '2',
  orange: '3',
  yellow: '4',
  green: '5',
  blue: '6',
  violet: '7',
  grey: '8',
  white: '9'
}.freeze

module ResistorColorDuo
  def self.value(colors)
    res = colors.map.reduce('') do |r, color|
      r + COLORS[color.to_sym]
    end
    res.to_i
  end
end
